tests/rules/changed_additional_properties_to_false_test.py

```diff
@@ -140,3 +140,53 @@ def test_validate_return_an_error(
             right_spec=new_spec,
         ),
     ) == expected_results
+
+
+def test_validate_succeeds_with_changes_to_additional_properties_objects(minimal_spec_dict, simple_operation_dict):
+    object_definitions = {
+        'ObjectThatRefersToAnotherObject': {
+            'additionalProperties': {'$ref': '#/definitions/AnotherObject'},
+            'type': 'object',
+        },
+        'AnotherObject': {
+            'type': 'object',
+            'properties': {
+                'property_1': {'type': 'string'}
+            },
+            'required': [
+                'property_1',
+            ]
+        },
+    }
+    old_spec_dict = dict(
+        minimal_spec_dict,
+        paths={
+            '/endpoint': {
+                'get': simple_operation_dict,
+            },
+        },
+        definitions=object_definitions,
+    )
+    old_spec_dict['paths']['/endpoint']['get']['parameters'] = [{
+        'in': 'body',
+        'name': 'body',
+        'required': True,
+        'schema': {
+            'properties': {
+                'property_1': {'$ref': '#/definitions/ObjectThatRefersToAnotherObject'}
+            },
+        },
+    }]
+    new_spec_dict = deepcopy(old_spec_dict)
+    new_spec_dict['definitions']['AnotherObject']['properties']['property_2'] = {'type': 'string'}
+    new_spec_dict['definitions']['AnotherObject']['required'].append('property_2')
+
+    old_spec = load_spec_from_spec_dict(old_spec_dict)
+    new_spec = load_spec_from_spec_dict(new_spec_dict)
+
+    assert list(
+        ChangedAdditionalPropertiesToFalse.validate(
+            left_spec=old_spec,
+            right_spec=new_spec,
+        ),
+    ) == []
```

Comment:
This PR is very compelling but makes me wonder if in this test scenario we should have a validation error (regardless of the validator).

I'm wondering so because on the old spec we consider valid an object like `{"property_1": {"property_1": "something
"}}` but using the new spec it would no longer be considered valid because `property_2` is now a required property of `AnotherObject`.

I do believe that this would produce a regression into the validation logic.

Still I have not a full insight of the CORESERV-10844 content, so I would leave Core Services forks make the final determination in here and decide if this PR should be accepted and/or if changes are needed.

PS: pre-commit tests are failing ... please fix them

Reply:
That's true, the test was probably a misleading example since it's not backwards incompatible, but should pass for this specific check (changing additional properties to false). I copied the existing test to `tests/rules/added_required_property_in_request_test.py` and it errors as expected, and modified the test in `tests/rules/changed_additional_properties_to_false_test.py` to make the properties no longer required to make it seem like a valid backwards incompatible change.

And yes let me fix those pre-commit errors. I didn't realize that it didn't manage to successfully install on my host

Reply:
That's correct ... I think that I was commenting on the wrong file ... mostly because in here there is no change with respct to `required`.

But still, `ChangedAdditionalPropertiesToFalse` should trigger if there are changes with respect to `additionalProperties` and in here I don't see changes with respect to it.
The new spec do modify `AnotherObject` by adding a new property ... no alterations on the additional properties 😕

Reply:
`additionalProperties` is being modified here after dereferencing the `$ref`s, and without the change in `swagger_spec_compatibility/rules/changed_additional_properties_to_false.py`, this test does fail. The case that's causing problems for us was due to `additionalProperties` behind several `$ref`s and I wanted to reproduce it in this test. I'll make another test case with a simpler `additionalProperties` object.